roboplot/core/stepper_control.py

```diff
@@ -49,16 +73,89 @@ def forwards(self) -> bool:
     def forwards(self, value: bool) -> None:
         self._motor.clockwise = value != self._invert_axis
 
+    @property
+    def is_homed(self):
+        return self._is_homed
+
+    def home(self) -> None:
+        """
+        Home the axis by driving into the limit switch and setting the current_location upon reaching it.
+
+        The home_position argument to Axis.__init__ controls the direction in which to home as well as the value set
+        upon reaching it.
+        """
+        self.forwards = self._home_position.forwards
+        if any([switch.is_pressed for switch in self._limit_switches]):
+            raise limit_switches.UnexpectedLimitSwitchError("Cannot home if switch is already pressed!")
+
+        hit_location = self._step_expecting_limit_switch()
+        while hit_location is None:
+            hit_location = self._step_expecting_limit_switch()
+
+        distance_moved_since_switch_pressed = self.current_location - hit_location
```

Comment:
Assume this is supposed to be more accurate than the intended 2mm. 
Comment : As we define the homing location to be the point at which the limit is hit and we have backed away from that location we need to adjust the current location accordingly. Note that the first distances are in relation to the unhomed starting point.